examples/arduino-uno/src/bin/uno-timer.rs

```diff
@@ -0,0 +1,136 @@
+#![no_std]
+#![no_main]
+#![feature(abi_avr_interrupt)]
+
+/*
+ This is derived from Rahix' comment to
+https://github.com/Rahix/avr-hal/issues/75
+and then modernized to account for API drift since 2020
+
+*/
+
+use arduino_hal::hal::port::PB5;
+use arduino_hal::port::mode::Output;
+use arduino_hal::port::Pin;
+use arduino_hal::prelude::*;
+use avr_device::atmega328p::tc1::tccr1b::CS1_A;
+use avr_device::atmega328p::TC1;
+use core::mem;
+use panic_halt as _;
+use ufmt::{uWrite, uwriteln};
+
+struct InterruptState<LED> {
+    blinker: Pin<Output, LED>,
+}
+
+static mut INTERRUPT_STATE: mem::MaybeUninit<InterruptState<PB5>> = mem::MaybeUninit::uninit();
+
+#[arduino_hal::entry]
+fn main() -> ! {
+    let dp = arduino_hal::Peripherals::take().unwrap();
+    let pins = arduino_hal::pins!(dp);
+
+    let mut serial = arduino_hal::default_serial!(dp, pins, 57600);
+    ufmt::uwriteln!(&mut serial, "Hello from Arduino!\r").void_unwrap();
+
+    let led = pins.d13.into_output();
+
+    unsafe {
+        // SAFETY: Interrupts are not enabled at this point so we can safely write the global
+        // variable here.  A memory barrier afterwards ensures the compiler won't reorder this
+        // after any operation that enables interrupts.
+        INTERRUPT_STATE = mem::MaybeUninit::new(InterruptState { blinker: led });
+        core::sync::atomic::compiler_fence(core::sync::atomic::Ordering::SeqCst);
+    }
+
+    //
+
+    let tmr1: TC1 = dp.TC1;
+
+    rig_timer(&tmr1, &mut serial);
+
+    // Enable interrupts globally, not a replacement for the specific interrupt enable
+    unsafe {
+        // SAFETY: Not inside a critical section and any non-atomic operations have been completed
+        // at this point.
+        avr_device::interrupt::enable();
+    }
+
+    if true {
+        ufmt::uwriteln!(
+            &mut serial,
+            "configured timer output compare register = {}",
+            tmr1.ocr1a.read().bits()
+        )
+        .void_unwrap();
+    }
+    loop {
+        if true {
+            avr_device::asm::sleep()
+        } else {
+            arduino_hal::delay_ms(100);
+
+            ufmt::uwriteln!(&mut serial, "counter = {}", tmr1.tcnt1.read().bits()).void_unwrap();
+        }
```

Comment:
Please drop the `if true {}` blocks and instead always do the print.  Let's keep the example code as simple as possible.

Reply:
I am having difficulty finding the right `use` clause to get the `Pin<Output,Dynamic>` to compile.  I'll keep banging on it.  So far I just have
```
struct InterruptState {
    blinker: Pin<Output, <PB5 as PinOps>::Dynamic>,
}
```

Reply:
Not sure what you're doing wrong...

```rust
struct InterruptState {
    blinker: arduino_hal::port::Pin<Output>,
}
```

should just work.  Maybe the problem is elsewhere?  Are you actually downgrading the pin before storing it in the static?

Reply:
Ah, the second parameter to the generic is optional...